src/librustdoc/json/conversions.rs

```diff
@@ -417,13 +417,13 @@ impl FromWithTcx<clean::Type> for Type {
                 }
             }
             Generic(s) => Type::Generic(s.to_string()),
+            Primitive(clean::PrimitiveType::Never) => Type::Never,
```

Comment:
Seems like the JSON format should be cleaned up to use `Type::Primitive("never")` instead of special-casing. Can you open an issue?

Reply:
Opened https://github.com/rust-lang/rust/issues/89349.